tests/conftest.py

```diff
@@ -68,6 +68,7 @@ def restore_env_variables():
         # set by XLA
         "XRT_MESH_SERVICE_ADDRESS",
         "XRT_TORCH_DIST_ROOT",
+        "TF2_BEHAVIOR",
```

Comment:
How is this related?

Reply:
I run one of our test and found this variable variable was leaking. 

If you run `pytest tests/trainer/connectors/test_checkpoint_connector.py::test_hpc_max_ckpt_version`, 

you would get 
```py
ERROR tests/trainer/connectors/test_checkpoint_connector.py::test_hpc_max_ckpt_version - AssertionError: test is leaking environment variable(s): {'TF2_BEHAVIOR'}
```

I guess it gets cleaned out by another test, but fails alone. So I added there. 

@awaelchli What is the intended purpose of this test ?

Reply:
It does pass for me, likely an issue in your environment.

The purpose is to track which env variables are leaking. With the intent to reduce the number and avoiding introducing new ones

Reply:
Yes, let me open an issue to track this later.

Reply:
> @awaelchli What is the intended purpose of this test ?

You were asking about test_hpc_max_ckpt_version?
It does what it says in the docstring and it does not set any env variables. I would have guessed maybe you have a newer tensorboard version and something gets triggered there. 
Also, I don't see how this env variable can be set by XLA if the test never runs on TPU